payment-request/rejects_if_not_active.https.html

```diff
@@ -22,68 +22,120 @@
-const validAmount = Object.freeze({
-  currency: "USD",
-  value: "5.00",
-});
-const validTotal = Object.freeze({
-  label: "Total due",
-  amount: validAmount,
-});
+
 const validDetails = Object.freeze({
-  total: validTotal,
+  total: {
+    label: "Total due",
+    amount: {
+      currency: "USD",
+      value: "5.00",
+    },
+  },
 });
 
 async function getLoadedPaymentRequest(iframe, url) {
   await new Promise((resolve) => {
     iframe.addEventListener("load", resolve, { once: true });
     iframe.src = url;
   });
-  const { PaymentRequest } = iframe.contentWindow;
-  const request = new PaymentRequest(validMethods, validDetails);
-  return request;
+  return new iframe.contentWindow.PaymentRequest(validMethods, validDetails);
 }
 
 promise_test(async (t) => {
   const iframe = document.createElement("iframe");
+  iframe.allow = "payment";
   document.body.appendChild(iframe);
-  // Make a request in the iframe.
-  const request = await getLoadedPaymentRequest(
+  // We first got to page1.html, grab a PaymentRequest instance.
+  const request1 = await getLoadedPaymentRequest(
     iframe,
-    "resources/page1.html"
+    "./resources/page1.html"
   );
-  await test_driver.bless("show payment request");
-  const showPromise = request.show();
-  const firstCtor = iframe.contentWindow.DOMException;
+  // Save the DOMException of page1.html before navigating away.
+  const frameDOMException1 = iframe.contentWindow.DOMException;
+
+  // Get transient activation
+  await test_driver.bless("payment 1", () => {}, iframe.contentWindow);
 
-  // Navigate the iframe to a new location. Wait for the load event to fire.
+  // We navigate the iframe again, putting request1's document into an non-fully active state.
   const request2 = await getLoadedPaymentRequest(
     iframe,
-    "resources/page2.html"
+    "./resources/page2.html"
   );
 
+  // Now, request1's relevant global object's document is no longer active.
+  // So, call .show(), and make sure it rejects appropriately.
   await promise_rejects_dom(
     t,
     "AbortError",
-    firstCtor,
-    showPromise,
-    "Show promise should abort."
+    frameDOMException1,
+    request1.show(),
+    "Inactive document, so must throw AbortError"
   );
-
-  // The navigation dismisses the previous payment request so it
-  // now possible to show another one.
-  await test_driver.bless("show 2nd payment request");
-  const showPromise2 = request2.show();
-  const secondCtor = iframe.contentWindow.DOMException;
-  // no longer fully active
+  // request2 has an active document tho, so confirm it's working as expected:
+  // Get transient activation
+  await test_driver.bless("payment 2", () => {}, iframe.contentWindow);
+  request2.show();
+  await request2.abort();
+  await promise_rejects_dom(
+    t,
+    "InvalidStateError",
+    iframe.contentWindow.DOMException,
+    request2.show(),
+    "Abort already called, so InvalidStateError"
+  );
+  // We are done, so clean up.
   iframe.remove();
```

Comment:
Nit; for future consideration, this might want to be in a cleanup function: https://web-platform-tests.org/writing-tests/testharness-api.html?highlight=cleanup#cleanup

Reply:
good suggestion. I didn't know about that.